[FIX] web_editor, mass_mailing: autosave image changes and snippet drops

Mtaylorr opened the pull request:
Issue:
======
Image and icons changes and dropped snippets are not autosaved, changes
are lost when you switch tab.

Steps to reproduce the issue:
=============================
For Image and icon:
- Go to email marketing
- Choose a template with an image
- Update the image and click on another tab directly (A/B tests for
  example)
- Go back to mail body tab, the changes are not saved

For dropped snippets:
- Go to email marketing
- Choose any template
- Click on another tab (A/B tests for example)
- Go back to meil body tab
- Add snippet click directly on another tab
- Go back to mail body tab , the changes are not saved

Origin of the issue:
====================
For Image and Icon:
When we open the image media dialog, `_onWysiwygBlur` is called which
means we lost the focus from the editable view. Old updates are saved
because we call `commitChanges` but the updates after the change of the
image are not since we lost the focus of the wysiwyg. Switching to another
tab will not trigger `blur` event again so we loose the changes. (same
flow for icon change).

For dropped snippets:
The first switch of tabs will trigger the blur event and will commit
changes. When we switch again to the mail body tab, the focus isn't
on the editable, we drop the snippet , still no focus on the editable
so no `blur` event is called and we loose the changes again.

We call `commitChanges` on `onWillUnmount` but we don't pass the
paremeters from `mass_mailing_html_field` to its parent class so we
loose the `urgent` flag which is responsible to save the data before
destroying the component. Passing just the args in `commitChanges` will
produce another issue about `Component is destroyed`. To have a minimal
change in stable we just refocus on the wysiwyg on the mentioned 2 cases
so the updates will be applied when blurring the `wysiwyg`

Solution:
=========
Put the focus again in the editor's window.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/167753).

Mtaylorr commented:
@dmo-odoo can you take a look at this when you have time ? thanks !

dmo-odoo commented:
I'm confused by this PR because there have already been prior commits that aimed to solve this issue of losing changes when switching tabs. Why did they not succeed? Were they not perfectly working in the first place or did the problem reappear for some reason due to another cause?

Mtaylorr commented:
@dmo-odoo , I updated the fix here as dicussed in private.
Note: :I added the urgentSave line because I have these 2 tests failing `test_02_mass_mailing_snippets_menu_tabs` and `test_04_mass_mailing_snippets_menu_hide` because of `component already destroyed` which we encountred before and its reason is the race condition where the component is destroyed before we call `updateValue`

dmo-odoo commented:
@robodoo r+
